docs/websites/varnish/use-varnish-with-nginx-to-serve-cached-wordpress-content-over-ssl-on-debian-8.md

```diff
@@ -0,0 +1,683 @@
+		~~~
+
+23. After saving and exiting the file you have to reload the systemd process:
+
+        sudo systemctl daemon-reload
+
+24. To make Varnish start at boot up, just run the following command:
+
+		sudo systemctl enable varnish.service
+
+We won't start Varnish at this point because we need to configure nginx first.
+
+## Install and Configure PHP
+
+Before configuring nginx we have to install PHP-FPM. FPM is short for FastCGI Process Manager, and it allows the web server to act as a proxy passing all requests with the `.php` file extension to the PHP interpreter.
+
+1. To install PHP-FPM just run:
+
+        sudo apt-get install php5-fpm php5-mysql
+
+2. Open the `/etc/php5/fpm/php.ini` file to configure PHP-FPM:
+
+		sudo nano /etc/php5/fpm/php.ini
+
+3. Find the directive `cgi.fix_pathinfo=` and set it to `0`, for safety reasons. If this parameter is set to `1`, the PHP interpreter will try to process the file who's path is as close to the requested path as possible, if it's set to `0`, the interpreter will only process the file with the exact path, which is a safer alternative.
+
+		cgi.fix_pathinfo=0
+
+4. Now open `www.conf`:
+
+		sudo nano /etc/php5/fpm/pool.d/www.conf
+
+5. Edit the `listen =` directive to specify the socket that will be used by nginx to pass requests to PHP-FPM:
+
+		listen = /var/run/php5-fpm.sock
+
+6.  Restart PHP-FPM:
+
+		sudo systemctl restart php5-fpm
+
+7.  Now open `/etc/nginx/fastcgi_params`:
+
+		sudo nano /etc/nginx/fastcgi_params
+
+8.  Find the line `fastcgi_param  HTTPS  $https if_not_empty;`. Below it add the following two lines, necessary for nginx to interact with the FastCGI service:
+
+	{: .file-excerpt }
+	/etc/nginx/fastcgi_params
+	:   ~~~ nginx
+
+		fastcgi_param  SCRIPT_FILENAME    $request_filename;
+		fastcgi_param  PATH_INFO          $fastcgi_path_info;
+		~~~
+
+	Save and exit the file.
+
+## Configure nginx
+
+1.  Open `/etc/nginx/nginx.conf`:
+
+		sudo nano /etc/nginx/nginx.conf
+
+2. Comment out the following two lines, because we'll include these SSL settings in the server block within the `/etc/nginx/sites-enabled/default` file:
+
+	{: .file-excerpt }
+	/etc/nginx/nginx.conf
+	:   ~~~ nginx
+
+		# ssl_protocols TLSv1 TLSv1.1 TLSv1.2; # Dropping SSLv3, ref: POODLE
+		# ssl_prefer_server_ciphers on;
+		~~~
+
+3. By default nginx makes use of an access log format called 'combined.' This type of logging is used by many web servers because it records all the information that is generally required by log analysis tools. The access log format can be changed from default to a custom one by using the `log_format` directive, but since we don't need custom logging we'll let nginx use the default 'combined' option.
+
+    Since the access logs and error logs will be defined for each individual website in the server block, you will also need to comment out the two lines below:
+
+	{: .file-excerpt }
+	/etc/nginx/nginx.conf
+	:   ~~~ nginx
+	
+		# access_log /var/log/nginx/access.log;
```

Comment:
Are these directives not overruled by those set in `/etc/nginx/sites-available/default`? Since we only define log paths for the https site, not the http one, aren't we leaving that site without logs? In other words, wouldn't it be safer to leave these uncommented as a catch for server blocks without log files defined?